### DISCUSSION TITLE
Can I use reloadDocument in useSubmit?

### DISCUSSION THREAD

**Discussion by fcosrno** [remix-run/remix#6339] [Q&A] [answered]
Can I use reloadDocument in useSubmit? I am working on a data table in Remix where clicking on a button posts to a Resource Route to download a dynamic csv file. This works if the submit button is inside the form with the reloadDocument prop, like this:

```typescript
// This works

<Form reloadDocument action="/export" method="post" >
    <input readOnly type="hidden" name="ids" value={selected.join(",")} />
    <button type="submit">Submit</button>
</Form>
```

But because of UI reasons, my submit button needs to live elsewhere, so I am using `useSubmit()` to submit the form after listening to state:

```typescript
// This does not work

const submit = useSubmit();
const [selected, setSelected] = React.useState<string[]>([]);
const [selectedAction, setSelectedAction] = React.useState<string>();
const selectedIdsFormRef = React.useRef<HTMLFormElement>(null);
React.useEffect(() => {
    if (selectedIdsFormRef.current && selectedAction !== undefined) {
        submit(selectedIdsFormRef.current, { replace: false });
    }
}, [selectedAction, submit]);

// Notice this button is outside the scope of the form and triggers the submit() in useEffect() above
<button onClick={()=>{setSelectedAction('send')}}>Submit</button>

// Notice reloadDocument is in this Form but the submit() function in useEffect() ignores this
<Form reloadDocument ref={selectedIdsFormRef} action="/export" method="post">
    <input readOnly type="hidden" name="ids" value={selected.join(",")} />
</Form>

```

By looking at the FormProps definitions I notice `reloadDocument?: boolean;` alongside `method?: FormMethod;`, `action?: string;`, `encType?: FormEncType;` and `replace?: boolean;`. The `SubmitOptions` for `useFormAction` also has `method`, `action`, `encType`, and `replace`, but no `reloadDocument`.

Is this a missing functionality or should I be doing this some other way?

Thanks!

**Comment by machour** [comment, marked as answer]
You can have your button outside your form without any problem:

```jsx
<form id="foo">
 ...
</form>

<button type="submit" form="foo" />
```

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#form

> The [<form>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) element to associate the button with (its form owner). The value of this attribute must be the id of a <form> in the same document. (If this attribute is not set, the <button> is associated with its ancestor <form> element, if any.)
>
> This attribute lets you associate <button> elements to <form>s anywhere in the document, not just inside a <form>. It can also override an ancestor <form> element.

Enjoy !

**Reply by fcosrno** [reply]
Ah, of course! This works! "Use the platform ™"
